### PR TITLE
Add FCS preview and indexing support

### PR DESCRIPTION
**Pull request by akarve**

### CODE REVIEW

File: lambdas/search/tests/test_search.py [hunk 1]
```diff
@@ -291,7 +291,7 @@ def _callback(request):
             assert payload['query']['query_string']
             assert payload['query']['query_string']['fields']
             assert payload['query']['query_string']['query']
-
+
```
Comment (akarve) [inline lambdas/search/tests/test_search.py:294]:
```suggestion

```

File: lambdas/shared/t4_lambda_shared/preview.py [hunk 2]
```diff
@@ -23,6 +26,7 @@
 ELASTIC_LIMIT_BYTES = int(os.getenv('DOC_LIMIT_BYTES') or 10_000)
 ELASTIC_LIMIT_LINES = 100_000
 MAX_PREVIEW_ROWS = 1_000
+TEMP_DIR = "/tmp"
```
Comment (akarve) [inline lambdas/shared/t4_lambda_shared/preview.py:29]:
```suggestion
TEMP_DIR = "/tmp"  # this is the default tmp directory in AWS Lambda
```

File: lambdas/shared/t4_lambda_shared/preview.py [hunk 3]
```diff
@@ -55,6 +59,50 @@ def decompress_stream(chunk_iterator, compression):
...
+            info - metdata for user consumption
+    """
+    meta = {}
+    data = None
+    body = ""
+    info = {}
+    # per Lambda docs we can use tmp/*, OK to overwrite
+    file_path  = os.path.join(TEMP_DIR, "quilt-temp.fcs")
+    # fcsparser only takes paths, so we need to write to disk; OK because
+    # FCS files typically ~1MB
```
Comment (kevinemoore) [inline lambdas/shared/t4_lambda_shared/preview.py:81]: This seems like a good tradeoff to get started, but FCS files can supposedly be large (e.g., the spec was updated to support sizes > 100MB).

Comment (akarve) [reply lambdas/shared/t4_lambda_shared/preview.py:81]: We have 500Mb of disk so we should be fine

File: lambdas/shared/t4_lambda_shared/preview.py [hunk 4]
```diff
@@ -55,6 +59,50 @@ def decompress_stream(chunk_iterator, compression):
...
+        dict
+            body - summary of main contents (if applicable)
+            info - metdata for user consumption
+    """
+    meta = {}
+    data = None
+    body = ""
+    info = {}
+    # per Lambda docs we can use tmp/*, OK to overwrite
+    file_path  = os.path.join(TEMP_DIR, "quilt-temp.fcs")
```
Comment (akarve) [inline lambdas/shared/t4_lambda_shared/preview.py:79]:
```suggestion
    file_path = os.path.join(TEMP_DIR, "quilt-temp.fcs")
```

File: lambdas/shared/t4_lambda_shared/preview.py [hunk 5]
```diff
@@ -55,6 +59,50 @@ def decompress_stream(chunk_iterator, compression):
...
+        if as_html:
+            body = remove_pandas_footer(data._repr_html_())  # pylint: disable=protected-access
+        # indexing
+        else:
+            body = ",".join(data.columns)
+    info['metadata'] = meta
+
+    return body, info
+
+
```
Comment (akarve) [inline lambdas/shared/t4_lambda_shared/preview.py:104]:
```suggestion
```

**Pull request merged**